packages/app/src/analytic-data-tracker/services/analytic-data-tracker.service.test.ts

```diff
@@ -128,13 +132,44 @@ describe("`AnalyticDataTrackerService`", () => {
     })
   })
 
+  describe("`externalUsageDevice` method", () => {
+    test("`setExternalUsageDevice` successfully set `externalUsageDevice` flag to `true`", async () => {
+      const subject = new AnalyticDataTrackerService(
+        { ...analyticDataTrackerOptions, trackingEnabled: true },
+        trackerCacheService,
+        axiosInstance
+      )
+      subject.setExternalUsageDevice(true)
+      axiosMock.onPost(apiUrl).reply(200)
+      const response = await subject.track({})
+      expect(response).not.toEqual(undefined)
+    })
+
+    test("`setExternalUsageDevice` successfully set `externalUsageDevice` flag to `false`", async () => {
+      const subject = new AnalyticDataTrackerService(
+        { ...analyticDataTrackerOptions, trackingEnabled: true },
+        trackerCacheService,
+        axiosInstance
+      )
+      subject.setExternalUsageDevice(true)
+      axiosMock.onPost(apiUrl).reply(200)
+      const response = await subject.track({})
+      expect(response).not.toEqual(undefined)
+
+      subject.setExternalUsageDevice(false)
+      const response2 = await subject.track({})
+      expect(response2).toEqual(undefined)
```

Comment:
How does this unit test check that the flag is set to false when the body of the logic seems to look similar?

Reply:
I don't see `not`. Sorry @lkowalczyk87